### ISSUE TITLE
application not take in charge

### ISSUE THREAD

**Issue by cedricvig** [Microsoft/vsts-rm-extensions#185]
We cannot add an application to a site without use advanced/Additional AppCmd.exe Commands.
The problem with the additional commands is that we cannot test if the app exists before delete or add it.
Is it possible to add a form that handle the app ?

**Comment by RoopeshNair** [comment]
Yes, we have in the backlog to support web applications and virtual folders support in addition to website updates. It would be available in ~2-3 months based on current priorities. Please stay tuned.

**Comment by Oladev** [comment]
We have one main IIS website, containing about 25-30 web applications, all having their own app pool. We develop locally using IIS Express, so we are not able to package IIS/application pool settings in the web package. Nor do we want to btw, because we want the deployment system to be in charge of the application pool being used in the dev/test/prod, not the individual developer.

During automatic deployment, we want to create an app pool for each application (if it does not exist), and install the application with the correct application pool. In the current version, this is hardly possible. Task 1 is ”IIS Web App Management” to create an app pool using the AppCmd.exe parameters, task 2 is “IIS Web App Deployment” to actually install the web application with its default app pool, and then task 3 is a ”IIS Web App Management” task to associate the installed application with the correct app pool. But since we can’t perform any conditional checks at application pool level (“does the application pool already exist”, etc.), it is very cumbersome to do it this way. Also, since we can’t check for existence or just simply update an app pool, we have to create the app pool before deployment. An error occurs when it already exists though, and the deployment process stops. So we tried to first delete the app pool, but then we receive an error if it does not exist and the deployment process stops as well.

We use this TFS extension currently for some TFS 2017 testing, but it seems completely unusable this way for the way we deploy. We just want to deploy several web applications under one main website, each one having their own application pool, and the IIS settings and application pool settings are not included in the web deployment package. I think it’s strongly related to the issue described here. I hope our problem is solved also once this issue is resolved.
